Doc(menu translation): application menu translation (#1367)
bonitasoft/bonita-doc: md/multi-language-applications.md

# Multi-language applications

> Throughout this document, you will see how we can translate the menus in Living Applications.

Reviewer: ```suggestion
Throughout this page, you will see how you can translate the application menus.
```

The menus in a custom application developed by a user can be available and extendable in the languages as per their requirements. But, the menus in Bonita living application are available in five different languages such as; English, French, Spanish, Japanese, and Portuguese (Brazilian) by default. 

> Based on the needs, the menus can be translated into other languages also by updating the Bonita layout.

Reviewer: I find the first sentence unclear. I think we could remove it.
also if we put this doc page in branch 7.12 of the doc this is not true that the provided apps menu are translated. It will be the case in 7.13 only and we need to update this page in 7.13 branch once this PR is merged. 
```suggestion
The menus in Bonita User and Administrator applications are available in English only by default. But, depending on your requirements, the menus can be translated into other languages too by updating Bonita layout. Using the same procedure, custom applications menus can be translated as well.  
```

> This documentation is only about the menus. For more information regarding translation at the page level, go to the [multi language pages](multi-language-pages.md) page.  

Reviewer: ```suggestion
This documentation is only about the menus. For more information regarding translations at the page level, go to the [multi language pages](multi-language-pages.md) page.  
```

## Updating the menu widget

>  In order to get the menu translated, we need to ensure that we have "| uiTranslate" added to the menu display name.    

Reviewer: ```suggestion
In order to get the menu translated, you need to ensure that the the `uiTranslate` directive is present for the menu entries display names in `customLayoutMenuBonitaV_` widget.    
```

>  You can follow the given procedures below. 

Reviewer: ```suggestion
In order to do so, you can follow this procedure:  
```

 
1. Update your widget inside Bonita UI Designer.
2. In the *Template* section, add "| uiTranslate" after *menu.displayName* and *childMenu.displayName* wherever it is. For example, *menu.displayName | uiTranslate*.
3. Save the widget.

![Menu-widget](images/multi-language-applications/menu-widget.png)

## Updating the layout

> A layout is a page used to define the main frame of an application.

Reviewer: I don't think we need to remind that here. However, I would rather say
```suggestion
Not that the translation mechanism has been added to the menu, you need to modify the layout to translate the menu entries. In order to do so, you can follow this procedure:  
```

1. Update your layout inside Bonita UI Designer. For more information regarding customizing the layout, go to the [customize layouts](customize-layouts.md) page.

> 
> 2. Inside the Localization section, click on the pencil icon on the right to the localizaton.json to edit it.
> ![Edit-localization](images/multi-language-applications/edit-localization.png)
> 3. The localization.json file shows four different sets of strings based on the languages available. 
> ![Localization](images/multi-language-applications/localization.png)
> 4. Now We need to add all the new keys in English which are the menu display names of the application you want to translate. For example, we will add the menu keys of Bonita User Application such as; Processes, Cases, and Tasks inside fr-FR, es-ES, jp-JP, and pt-BR objects.
> ![Add-keys](images/multi-language-applications/add-keys.png)
> 5. Click on **Save** to save the changes we made in the localization.json.
> 6. Click on **Close**.
> 7. Now save the layout.
> 8. Now, deploy the layout on your target runtime.

Reviewer: ```suggestion
1. Make sure your layout uses the version of the menu widget you just updated (click on the menu widget in the whiteboard to check its name and version and use the replace feature at the top right corner of the properties panel it is not the right version).
1. Inside the Localization section of the assets panel, click on the pencil icon at the right of the localizaton.json file to edit it.
![Edit-localization](images/multi-language-applications/edit-localization.png)
1. The localization.json file shows four different sets of strings based on the languages available. 
![Localization](images/multi-language-applications/localization.png)
1. Now We need to add all the new keys in English which are the menu entry display names for the application you want to translate. For example, we will add the menu entry keys of Bonita User Application such as Processes, Cases, and Tasks inside fr-FR, es-ES, jp-JP, and pt-BR objects.
![Add-keys](images/multi-language-applications/add-keys.png)
1. Click on **Save** to save the changes we made in the localization.json file.
1. Click on **Close**.
1. Save and export the layout.
1. Now, deploy the layout on your target runtime and use it in your applications.
```
